hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -467,22 +467,27 @@ private void removeTrigger(Trigger trigger) {
             // then remove everything else.
             // We may want to do this with rules, because as is, we need to loop through every Fact in
             // the rules engine doing a relatively slow check.
+            final String tenantId = trigger.getTenantId();
             final String triggerId = trigger.getId();
             rules.removeFacts(t -> {
                 if (t instanceof Dampening) {
-                    return ((Dampening) t).getTriggerId().equals(triggerId);
+                    return ((Dampening) t).getTenantId().equals(tenantId)
+                            && ((Dampening) t).getTriggerId().equals(triggerId);
                 } else if (t instanceof Condition) {
-                    return ((Condition) t).getTriggerId().equals(triggerId);
+                    return ((Condition) t).getTenantId().equals(tenantId) &&
+                            ((Condition) t).getTriggerId().equals(triggerId);
                 }
```

Comment:
Wow, yes, this was definitely a problem. Good catch.  I still wonder whether we should do this in rules to simplify the java side.